File: packages/mobile/lib/domain/usecases/authentication/signup_usecase.dart
import 'package:entrepoty/domain/domain.dart';
import 'package:fpdart/fpdart.dart';

abstract class SignupUsecase {
  Future<Either<Account, DomainError>> register({
    required String name,
    required String username,
    required String password,
  });
}

File: packages/server/src/main/kotlin/com/entrepoty/Entrepoty/domain/usecase/SignupUsecase.kt
package com.entrepoty.Entrepoty.domain.usecase

import arrow.core.Either
import com.entrepoty.Entrepoty.domain.entities.AuthenticationModel
import com.entrepoty.Entrepoty.domain.entities.CreateAccountModel
import com.entrepoty.Entrepoty.domain.entities.DomainError

interface SignupUsecase {
    fun registerAccount(account: CreateAccountModel): Either<DomainError, AuthenticationModel>;
}